packages/synapse-react-client/src/components/FeaturedDataTabs/FacetPlotsCard.tsx

```diff
@@ -95,36 +91,39 @@ function LoadingCard(props: { numPlots: number }) {
     </Paper>
   )
 }
-const FacetPlotsCard: React.FunctionComponent<FacetPlotsCardProps> = ({
-  title,
-  description,
-  facetsToPlot,
-  detailsPagePath,
-}: FacetPlotsCardProps): JSX.Element => {
+
+function FacetPlotsCard(props: FacetPlotsCardProps) {
+  const { title, description, facetsToPlot, detailsPagePath } = props
   const { accessToken } = useSynapseContext()
-  const isLoadingNewBundle = useAtomValue(isLoadingNewBundleAtom)
-  const data = useAtomValue(tableQueryDataAtom)
+  const { queryMetadataQueryOptions } = useQueryContext()
+  const { data: queryMetadata } = useSuspenseQuery(queryMetadataQueryOptions)
 
   const { getColumnDisplayName } = useQueryVisualizationContext()
-  const [facetPlotDataArray, setFacetPlotDataArray] = useState<GraphData[]>([])
-  const [facetDataArray, setFacetDataArray] = useState<FacetColumnResult[]>([])
-  const [selectedFacetValue, setSelectedFacetValue] = useState<string>('')
 
-  useEffect(() => {
```

Comment:
This `useEffect` set the following state variables: `facetPlotDataArray`, `facetDataArray`, and `selectedFacetValue`.

 Simplify/break this apart:

`facetDataArray` and `selectedFacetValue` can each be created by in their own `useMemo`. `facetPlotDataArray` requires multiple asynchronous calls, so wrap it in a `useSuspenseQuery` and the component will just suspend until it's done.